core/src/main/java/org/apache/iceberg/BaseMetadataTableScan.java

```diff
@@ -38,4 +44,14 @@ public long targetSplitSize() {
         TableProperties.METADATA_SPLIT_SIZE_DEFAULT);
     return PropertyUtil.propertyAsLong(options(), TableProperties.SPLIT_SIZE, tableValue);
   }
+
+  /**
+   * Alternative to {@link #planFiles()}, allows exploring old snapshots even for empty table.
+   */
+  protected CloseableIterable<FileScanTask> planAllFiles() {
```

Comment:
Could this be in BaseFilesTableScan? Just wondering if it would be relevant for say "Manifests" tables?

Reply:
Yea this was tricky and I went back and forth with various solutions.  

BaseAllMetadataTableScan used to override planFiles() , and was inherited by AllDataFiles, AllManifests, AllEntries.  

Now I moved AllDataFiles to BaseFilesTableScan and so I don't get this override anymore.

So it has to be called from two different hierarchies (BaseAllMetadataTableScans and BaseFilesTableScan), hence moving to their only common parent to continue sharing the code.

Reply:
Actually I tried to improve a little bit and made the override at the BaseMetadataTableScan parent class, based on a boolean flag the subclasses configure, let me know if that is better.